src/compiler/scanner.ts

```diff
@@ -605,12 +610,23 @@ module ts {
         }
 
         function isIdentifierStart(ch: number): boolean {
+            if (inJSXTag && ch === CharacterCodes.backslash) {
+                return false;
+            }
             return ch >= CharacterCodes.A && ch <= CharacterCodes.Z || ch >= CharacterCodes.a && ch <= CharacterCodes.z ||
                 ch === CharacterCodes.$ || ch === CharacterCodes._ ||
                 ch > CharacterCodes.maxAsciiCharacter && isUnicodeIdentifierStart(ch, languageVersion);
         }
 
         function isIdentifierPart(ch: number): boolean {
+            if (inJSXTag) {
+                if (ch === CharacterCodes.backslash) {
+                    return false;
+                } 
+                else if (ch === CharacterCodes.minus) {

```

Comment:
I would not do things this way.  Instead, we should either go the 'rescanXXX' route (like we do for regex/templates), or we should just scan these out a sequence of 'identifier + minus ...' tokens that the parser can accept as the name of an xml tag.